src/dtgtk/culling.c

```diff
@@ -1625,35 +1719,36 @@ static gboolean _thumbs_compute_positions(dt_culling_t *table)
                              // constructed list, so move it back to
                              // the start
 
-  total_width -= distance;
-  total_height -= distance;
+  // once we placed all images, we can remove the space at the right and the bottom
+  total_width -= spacing;
+  total_height -= spacing;
 
+  //int thumb_iter_counter = 0;
   for(const GList *iter = rows; iter; iter = g_list_next(iter))
   {
     GList *row = (GList *)iter->data;
-    int row_w = 0, xoff;
-    int max_rh = 0;
+    int row_width = 0, xoff;
+    int max_row_heigth = 0;
 
     for(GList *slot_cw_iter = row;
         slot_cw_iter;
         slot_cw_iter = g_list_next(slot_cw_iter))
     {
       dt_thumbnail_t *cw = (dt_thumbnail_t *)slot_cw_iter->data;
-      row_w = MAX(row_w, cw->x + cw->width);
-      max_rh = MAX(max_rh, cw->height);
+      row_width = MAX(row_width, cw->x + cw->width);
+      max_row_heigth = MAX(max_row_heigth, cw->height); //tbd: this is wrong for stacked images, should be (cw->y - row_y + cw->heigth)
```

Comment:
This tbd comment hint to some wrong code and propose a fix, why not put the fix now?

(same comment for the tbd comment below).

Reply:
I implemented the fix in the comment but I needed to rewrite some of the loops completely and even after that it did not resolve all issues. To keep this PR readable I decided to keep that fix out and publish it once I managed to figure out what is going wrong.
I will fix the other comments.

Reply:
This is what it looks like. Thumbnails overlay each other and leave the screen.
![image](https://github.com/darktable-org/darktable/assets/2805109/4807af2d-cde1-43ba-b832-167ec34fa348)

Reply:
Since nobody pointed this out so far, I don't think it has been encountered often

Reply:
> Since nobody pointed this out so far, I don't think it has been encountered often

I would say that if it is easy to fix and won't introduce regressions we should fix it now.

Reply:
Like I said, I have not fully identified the cause yet because it is not that simple. But I can already tell that I will need to make bigger changes to this function than what I already did.

If you do not mind that this PR grows even more complex, I can work on it. Otherwise we should split it into 2 PRs

Reply:
Understood, so let's merge this as it is already a big improvement. Thanks.